packages/puppeteer-core/src/util/disposable.ts

```diff
@@ -145,15 +143,40 @@ export class DisposableStack {
    */
   move(): DisposableStack {
     if (this.#disposed) {
-      throw new ReferenceError('a disposed stack can not use anything new'); // step 3
+      throw new ReferenceError('A disposed stack can not use anything new');
     }
-    const stack = new DisposableStack(); // step 4-5
+    const stack = new DisposableStack();
     stack.#stack = this.#stack;
```

Comment:
We need to nullify the stack here after copying it.
```suggestion
    stack.#stack = this.#stack;
    this.#stack = [];
```

Reply:
Fixed